agents/registry/registry.py

```diff
@@ -143,17 +146,21 @@ def main(self, session: ocs_agent.OpSession, params=None):
             }
 
             for addr, agent in self.registered_agents.items():
-                _addr = addr.replace('.', '_')
-                _addr = _addr.replace('-', '_')
-                msg = {
-                    'block_name': _addr,
-                    'timestamp': time.time(),
-                    'data': {
-                        f'{_addr}_{op_name}': op_code
-                        for op_name, op_code in agent.op_codes.items()
-                    }
-                }
-                self.agent.publish_to_feed('agent_operations', msg)
+                msg = { 'block_name': addr,
+                        'timestamp': time.time(),
+                        'data': {}}
+                for op_name, op_code in agent.op_codes.items():
+                    field = f'{addr}_{op_name}'
+                    field = field.replace('.', '_')
+                    field = field.replace('-', '_')
+                    try:
+                        Feed.verify_data_field_string(field)
+                    except ValueError as e:
+                        self.log.warn(f"Improper field name: {field}\n{e}")
+                        continue
```

Comment:
So agent address + op_name fields that are invalid are simply dropped? This is tricky to handle I suppose because an Agent could come online at any time with an address that could make this invalid. What's likely to happen is the Agent is brought up, and no operation status shows up for the user when expected, with no feedback on their end (since it's in the Registry.) This is the same problem we had with the Aggregator generating errors that would go unnoticed when new Agents would come online. Checking this combination of addr + op_name on the Agent heartbeat side would provide the user starting up the Agent feedback and prevent an invalid combo from getting this far.

The field name rules could also be enforced for invalid names. The Provider class does this for instances where older Agents are running on the newer Aggregator: https://github.com/simonsobs/ocs/blob/bba8a3f3378bee0eb9a6558f848579366b0d4570/ocs/agent/aggregator.py#L227

Reply:
I think using `_enforce_field_name_rules` is a good solution, I'll do that instead of adding registry-specific warnings to the `register_task` and `register_process` OCSAgent functions.